core/src/main/java/kafka/server/share/ShareSessionContext.java

```diff
@@ -0,0 +1,243 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server.share;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NoSuchElementException;
+import kafka.server.share.SharePartitionManager.ErroneousAndValidPartitionData;
+import org.apache.kafka.common.TopicIdPartition;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.message.ShareFetchResponseData;
+import org.apache.kafka.common.message.ShareFetchResponseData.PartitionData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.requests.ShareFetchMetadata;
+import org.apache.kafka.common.requests.ShareFetchRequest;
+import org.apache.kafka.common.requests.ShareFetchRequest.SharePartitionData;
+import org.apache.kafka.common.requests.ShareFetchResponse;
+import org.apache.kafka.server.share.CachedSharePartition;
+import org.apache.kafka.server.share.ShareSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+/**
+ * The context for a share session fetch request.
+ */
+public class ShareSessionContext extends ShareFetchContext {
+
+    private final ShareFetchMetadata reqMetadata;
+    private Map<TopicIdPartition, SharePartitionData> shareFetchData;
+    private final boolean isSubsequent;
+    private ShareSession session;
+
+    private final Logger log = LoggerFactory.getLogger(ShareSessionContext.class);
+
+    /**
+     * The share fetch context for the first request that starts a share session.
+     *
+     * @param reqMetadata        The request metadata.
+     * @param shareFetchData     The share partition data from the share fetch request.
+     */
+    public ShareSessionContext(ShareFetchMetadata reqMetadata,
+                               Map<TopicIdPartition, ShareFetchRequest.SharePartitionData> shareFetchData) {
+        this.reqMetadata = reqMetadata;
+        this.shareFetchData = shareFetchData;
+        this.isSubsequent = false;
+    }
+
+    /**
+     * The share fetch context for a subsequent request that utilizes an existing share session.
+     *
+     * @param reqMetadata  The request metadata.
+     * @param session      The subsequent fetch request session.
+     */
+    public ShareSessionContext(ShareFetchMetadata reqMetadata, ShareSession session) {
+        this.reqMetadata = reqMetadata;
+        this.session = session;
+        this.isSubsequent = true;
+    }
+
+    public Map<TopicIdPartition, ShareFetchRequest.SharePartitionData> shareFetchData() {
```

Comment:
Is it just declared for tests? Can we have `//Visible for testing` written for the method? Same for others.